src/check_error.c

```diff
@@ -18,6 +18,17 @@
  * MA 02110-1301, USA.
  */
 
+/**
+ * We need the GNU C version to prevent using pragmas where they are
+ * not supported.
+ */
+#define GCC_VERSION (__GNUC__ * 10000 \
```

Comment:
Should this be in libcompat instead? Or, is one goal to reduce/remove that header eventually?